zandronum fails to build on NixOS 16.03

## Issue description
zandronum fails to build on NixOS 16.03
Pinging maintainer @Lassulus 

## Steps to reproduce
`$ nix-shell -p zandronum`

## Error message
```
[ 15%] Building C object dumb/CMakeFiles/dumb.dir/src/core/atexit.o
In file included from /tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/src/core/atexit.c:22:0:
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:403:1: error: invalid use of 'restrict'
 DUH *DUMBEXPORT dumb_load_mod(const char *filename, int restrict);
 ^
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:403:1: error: invalid use of 'restrict'
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:416:1: error: invalid use of 'restrict'
 DUH *DUMBEXPORT dumb_read_mod(DUMBFILE *f, int restrict);
 ^
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:416:1: error: invalid use of 'restrict'
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:429:1: error: invalid use of 'restrict'
 DUH *DUMBEXPORT dumb_load_mod_quick(const char *filename, int restrict);
 ^
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:429:1: error: invalid use of 'restrict'
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:442:1: error: invalid use of 'restrict'
 DUH *DUMBEXPORT dumb_read_mod_quick(DUMBFILE *f, int restrict);
 ^
/tmp/nix-build-zandronum-2.1.2.drv-0/hg-archive/dumb/include/dumb.h:442:1: error: invalid use of 'restrict'
dumb/CMakeFiles/dumb.dir/build.make:54: recipe for target 'dumb/CMakeFiles/dumb.dir/src/core/atexit.o' failed
make[2]: *** [dumb/CMakeFiles/dumb.dir/src/core/atexit.o] Error 1
CMakeFiles/Makefile2:499: recipe for target 'dumb/CMakeFiles/dumb.dir/all' failed
make[1]: *** [dumb/CMakeFiles/dumb.dir/all] Error 2
Makefile:72: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/885zf31v51qkxgdzgrsgpkxxlr4mh2ls-zandronum-2.1.2.drv’ failed with exit code 2
error: build of ‘/nix/store/885zf31v51qkxgdzgrsgpkxxlr4mh2ls-zandronum-2.1.2.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

## Technical details
* System: 16.03.887.009fa8e (Emu)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: 16.03.887.009fa8e

--- a/pkgs/games/zandronum/default.nix
+++ b/pkgs/games/zandronum/default.nix
@@ -24,6 +24,8 @@ in stdenv.mkDerivation {
 
   preConfigure = ''
     ln -s ${sqlite-amalgamation}/* sqlite/
+    sed -ie 's| restrict| _restrict|g' dumb/include/dumb.h \
+                                       dumb/src/it/*.c
   '';
 
   cmakeFlags =

